lib/matplotlib/figure.py

```diff
@@ -1990,7 +1990,7 @@ def _do_layout(gs, mosaic, unique_ids, nested):
             ----------
             gs : GridSpec
             mosaic : 2D object array
-                The input converted to a 2D numpy array for this level.
+                The input converted to a 2D array for this level.
```

Comment:
This is clearly array-like as it takes lists.  Not sure if it accepts any array-like?

Reply:
Actually this is only ever called internally and at that point mosaic has indeed been converted to a numpy array (by _make_array).